src/hotspot/cpu/riscv/riscv_v.ad

```diff
@@ -363,6 +373,21 @@ instruct vadd(vReg dst, vReg src1, vReg src2) %{
   ins_pipe(pipe_slow);
 %}
 
+instruct vadd_hfp(vReg dst, vReg src1, vReg src2) %{
+  match(Set dst (AddVHF src1 src2));
+  ins_cost(VEC_COST);
+  format %{ "vadd_hfp $dst, $src1, $src2" %}
+  ins_encode %{
+    assert(UseZvfh, "must");
+    BasicType bt = Matcher::vector_element_basic_type(this);
```

Comment:
Question: What is `bt` calculated here? Seems there isn't one for HF16 in `enum BasicType` definition in file src/hotspot/share/utilities/globalDefinitions.hpp. I only see `T_FLOAT` and `T_DOUBLE`, which I don't think is usable here as we need to set SEW=16 for this instruction.

Reply:
No, it uses T_SHORT instead, in Float16.java it also uses a short as underlying payload.
And if you check the generated assembly code, you'll find some code like `vsetivli        t0,16,e16,m1,tu,mu`.

To avoid confusion, I will add an assertion here so that it can be understood later.

Reply:
Thanks for the answer. I see that is also reflected on the C2 source code [1].
Why not save this `Matcher::vector_element_basic_type(this)` call then? I mean:
```
assert(Matcher::vector_element_basic_type(this) == T_SHORT, "must");
__ vsetvli_helper(T_SHORT, Matcher::vector_length(this));
```

[1] https://github.com/openjdk/jdk/blob/master/src/hotspot/share/opto/vectornode.cpp#L63

Reply:
Make sense, fixed.